osu.Game/Screens/Play/ModsContainer.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using osu.Framework.Graphics.Containers;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.UI;
+using osu.Framework.Graphics;
+using OpenTK.Graphics;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+
+namespace osu.Game.Screens.Play
+{
+    public class ModsContainer : Container
+    {
+        private readonly FillFlowContainer<ModIcon> iconsContainer;
+
+        public bool ShowMods
+        {
+            set { if (!value) Hide(); }
+        }
+
+        public ModsContainer()
+        {
+            Children = new Drawable[]
+            {
+                iconsContainer = new FillFlowContainer<ModIcon>
+                {
+                    Anchor = Anchor.TopCentre,
+                    Origin = Anchor.TopCentre,
+                    AutoSizeAxes = Axes.Both,
+                    Direction = FillDirection.Horizontal,
+                },
+                new OsuSpriteText
+                {
+                    Anchor = Anchor.BottomCentre,
+                    Origin = Anchor.TopCentre,
+                    Text = @"/UNRANKED/",
+                    Font = @"Venera",
+                    TextSize = 15,
+                }
+            };
+        }
+
+        public void Add(Mod mod)
+        {
+            iconsContainer.Add(new ModIcon
+            {
+                AutoSizeAxes = Axes.Both,
+                Icon = mod.Icon,
+                Colour = selectColour(mod),
+                IconSize = 60,
+            });
+        }
+
+        private Color4 selectColour(Mod mod)

```

Comment:
not literally. as in a function to provide icon colours is necessary in multiple places.
it should probably be inside the mod drawable so it can be used in a single place for all cases